maitake/src/scheduler.rs

```diff
@@ -888,7 +888,11 @@ impl Core {
             has_remaining: false,
         };
 
-        for task in self.run_queue.consume().take(n) {
+        for _ in 0..n {
+            let Some(task) = self.run_queue.dequeue() else {
```

Comment:
one note about this is that `MpscQueue::dequeue` [will spin](https://docs.rs/cordyceps/latest/cordyceps/mpsc_queue/struct.MpscQueue.html#method.dequeue) if another core/thread is currently dequeuing, which...I don't love. I think the scheduler will still *work* if multiple cores tick it concurrently, but it will do a lot of spinning, compromising performance.

We may want to do one of the following:
 - add an `AtomicBool` or something that tracks whether the scheduler is being ticked, and have `tick` return immediately if it is set (to prevent concurrent ticks),
 - change this to use [`MpscQueue::try_dequeue`](https://docs.rs/cordyceps/latest/cordyceps/mpsc_queue/struct.MpscQueue.html#method.try_dequeue) and have the function bail if the queue is busy.

Reply:
if we want to use `try_dequeue`, I think it would make sense to change the loop to something like this:
```rust
let mut polled = 0;
while polled < n {
    let task = match self.run_queue.try_dequeue() {
         Ok(t) => t,
         // the queue is in an inconsistent state, just try again.
         Err(TryDequeueError::Inconsistent) => {
             core::hint::spin_loop();
             continue;
         }
         // the queue is empty, or another thread is ticking the scheduler. just bail out!
         Err(_) => break,
    };

    // actually poll the task ...

    polled += 1
}
```
changing this to a `while` loop lets us use the main loop body to retry inconsistent states in dequeue, without incrementing the polled tasks counter (which, when using the `for` loop, would always be incremented).